File: apps/cargo-tracker/app/cargotracker/estimation/domain/EstimateId.scala
package cargotracker.estimation.domain

import java.util.UUID

/** 見積の一意識別子。UUID ベース。 */
opaque type EstimateId = String

object EstimateId:
  /** 新しい EstimateId を発行する。 */
  def generate(): EstimateId = UUID.randomUUID().toString

  /** 永続化からの復元 */
  def unsafeFrom(raw: String): EstimateId = raw

  extension (id: EstimateId) def value: String = id

File: apps/cargo-tracker/src/main/java/com/example/cargotracker/estimation/domain/model/EstimateId.java
package com.example.cargotracker.estimation.domain.model;

import java.util.UUID;

/**
 * 見積 ID（US01 の受入基準 4「見積番号が発行される」）。
 *
 * @param value UUID
 */
public record EstimateId(UUID value) {

    public EstimateId {
        if (value == null) {
            throw new IllegalArgumentException("見積 ID は必須です");
        }
    }

    /** 新しい見積 ID を採番する。 */
    public static EstimateId generate() {
        return new EstimateId(UUID.randomUUID());
    }

    /** 文字列から復元する。 */
    public static EstimateId of(String value) {
        return new EstimateId(UUID.fromString(value));
    }

    @Override
    public String toString() {
        return value.toString();
    }
}
